src/open_inwoner/accounts/views/profile.py

```diff
@@ -106,10 +107,24 @@ def get_context_data(self, **kwargs):
     def post(self, request, *args, **kwargs):
         if request.user.is_authenticated and not request.user.is_staff:
             instance = User.objects.get(id=request.user.id)
-            self.request.user.deactivate()
 
+            # check if there are still plans associated witht the user
+            if Plan.objects.filter(created_by=instance).exists():
```

Comment:
Thanks for pointing this out. However, `Plan.objects.connected(user=instance)` only returns the plans for which the user is a contact, not the ones which were created by the user. I used a disjunctive query instead (`Plan.objects.filter(Q(created_by=instance) | Q(plan_contacts=instance))`).

Reply:
@vaszig I understand now. At the moment however checking whether `Plan.objects.connected(user)` will only work if the `Plan` instance is created by submitting the appropriate form. If you do `PlanFactory.create(created_by=user)` the test fails b/c the creator of the plan is only added at the view level. Wouldn't it be better to override the `create` method of the `PlanQuerySet` instead of doing it inside the view (as per the second link you shared)? Otherwise I can modify the test to accomodate this.

Reply:
I realized that the addition of plan creators to contacts is already tested elsewhere, so there's no need for that test case in the present ticket.